### ISSUE TITLE
Sentry causes `java.lang.NoSuchFieldError: No instance field mDelegate` with Room 2.5.0

### ISSUE THREAD

**Issue by twwd** [getsentry/sentry-android-gradle-plugin#434]
### Integration

sentry-android

### Build System

Gradle

### AGP Version

7.4.0

### Proguard

Enabled

### Version

probably latest (automatically added through io.sentry.android.gradle 3.2.0 or 3.4.1))

### Steps to Reproduce

1. Add Sentry to an Android app with Room 2.5.0
2. Insert something into the database

### Expected Result

Something is stored in the database/no exceptions

### Actual Result

```
java.lang.NoSuchFieldError: No instance field mDelegate of type Landroid/database/sqlite/SQLiteStatement; in class Landroidx/sqlite/db/framework/FrameworkSQLiteStatement; or its superclasses (declaration of 'androidx.sqlite.db.framework.FrameworkSQLiteStatement' appears in /data/app/~~pAvZq98N6TMS64i4kOmPzw==/de.visualvest.pushtan.app-sGrKnNFuWTsxX0Le4FBKJg==/base.apk!classes14.dex)
at androidx.sqlite.db.framework.FrameworkSQLiteStatement.executeInsert(Unknown Source:4)
at androidx.room.EntityInsertionAdapter.insert(EntityInsertionAdapter.kt:68)
``` 

Details see https://issuetracker.google.com/issues/265824784
Sentry seems to override the SQlite version

**Comment by twwd** [comment]
Seems to be caused by the bytecode manipulation from the tracing instrumentation. When I [disable](https://docs.sentry.io/platforms/android/configuration/integrations/room-and-sqlite/#configure), it, the app works with the upgrade Room version.

**Comment by romtsn** [comment]
Hi @twwd, apologies for the problems the plugin has caused. I'm pretty sure it's our instrumentation that breaks it, so we'll look into this asap. 

I have already filed https://github.com/getsentry/sentry-android-gradle-plugin/issues/401 a while ago to look into this and additionally set up a compatibility test matrix on CI, but, unfortunately, we never got to this and Room 2.5.0 has been released as stable meanwhile. 

The workaround for now is to either downgrade the Room version, or to disable tracing instrumentation as you mentioned:
```gradle
import io.sentry.android.gradle.extensions.InstrumentationFeature

sentry {
  tracingInstrumentation {
    enabled = true
    features = EnumSet.allOf(InstrumentationFeature) - InstrumentationFeature.DATABASE
  }
}
```

cc @kahest @zoesyc for visibility

**Comment by romtsn** [comment]
hey @twwd the fix has been released in v3.4.2, please try it out and let us know if there are still issues. Thanks!

**Comment by twwd** [comment]
Thanks for this ultra fast fix! I want to try it but I ran into #436 :(

**Comment by twwd** [comment]
It seems to work now :+1:
